tests/drivers/build_all/pwm/testcase.yaml

```diff
@@ -1,8 +1,74 @@
 common:
   build_only: true
-  tags: drivers
+  tags: drivers pwm
 tests:
+  drivers.pwm.gecko.build:
+    platform_allow: efr32_radio_brd4250b
+    tags: pwm_gecko
+    extra_args: "CONFIG_PWM_GECKO=y"
+  drivers.pwm.imx.build:
+    platform_allow: colibri_imx7d_m4
+    tags: pwm_imx
+    extra_args: "CONFIG_PWM_IMX=y"
+  drivers.pwm.litex.build:
+    platform_allow: litex_vexriscv
+    tags: pwm_litex
+    extra_args: "CONFIG_PWM_LITEX=y"
+  drivers.pwm.mcux.ftm.build:
+    platform_allow: frdm_k22f
+    tags: pwm_mcux_ftm
+    extra_args: "CONFIG_PWM_MCUX_FTM=y"
+  drivers.pwm.mcux.pwt.build:
+    platform_allow: twr_ke18f
+    tags: pwm_mcux_pwt
+    extra_args: "CONFIG_PWM_MCUX_PWT=y CONFIG_PWM_CAPTURE=y"
+  drivers.pwm.mcux.ftm.build:
+    platform_allow: frdm_kw41z
+    tags: pwm_mcux_ftm
+    extra_args: "CONFIG_PWM_MCUX_TPM=y"
+  drivers.pwm.mcux.build:
+    platform_allow: mimxrt1064_evk
+    tags: pwm_mcux
+    extra_args: "CONFIG_PWM_MCUX=y"
+  drivers.pwm.rv32m1.tpm.build:
+    platform_allow: rv32m1_vega_ri5cy
+    tags: pwm_rv32m1_tpm
+    extra_args: "CONFIG_PWM_RV32M1_TPM=y"
+  drivers.pwm.sifive.build:
+    platform_allow: hifive1_revb
+    tags: pwm_sifive
+    extra_args: "CONFIG_PWM_SIFIVE=y"
+  drivers.pwm.npcx.build:
+    platform_allow: npcx7m6fb_evb
+    tags: pwm_npcx
+    extra_args: "CONFIG_PWM_NPCX=y"
+  drivers.pwm.nrf.sw.build:
+    platform_allow: nrf52833dk_nrf52820
```

Comment:
Please use `nrf51dk_nrf51422` instead, as this driver is mostly used on nRF51, where the hardware PWM is not available.